src/main/java/org/twilmes/sql/gremlin/processor/executors/QueryExecutor.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.twilmes.sql.gremlin.processor.executors;
+
+import lombok.AllArgsConstructor;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+import org.twilmes.sql.gremlin.SqlToGremlin;
+import org.twilmes.sql.gremlin.schema.SchemaConfig;
+import org.twilmes.sql.gremlin.schema.TableDef;
+import org.twilmes.sql.gremlin.schema.TableRelationship;
+import org.twilmes.sql.gremlin.schema.TableUtil;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public abstract class QueryExecutor {
+    private static final int DEFAULT_PAGE_SIZE = 1000;
+    private static final int pageSize = DEFAULT_PAGE_SIZE;
```

Comment:
Weren't we just going to make this a private int since we may want to change it across different threads?

Reply:
Yes, will do that.